ibis/backends/sql/rewrites.py

```diff
@@ -244,6 +251,48 @@ def merge_select_select(_, **kwargs):
     if _.parent.find_below(blocking, filter=ops.Value):
         return _
 
+    if _.parent.distinct:
+        # The inner query is distinct.
+        #
+        # If the outer query is distinct, it's only safe to merge if it's a simple subselection:
+        # - Fusing in the presence of non-deterministic calls in the select would lead to
+        #   incorrect results
+        # - Fusing in the presence of expensive calls in the select would lead to potential
+        #   performance pitfalls
+        if _.distinct and not all(
+            isinstance(v, ops.Field) for v in _.selections.values()
```

Comment:
Does this cover the alias-in-outer-project case?

```sql
SELECT a, b AS c
FROM (
  SELECT DISTINCT
    a, b
  FROM t
)
```

would become

```sql
SELECT DISTINCT
  a, b AS c
FROM t
```

If not, fine to either handle later or perhaps never if it doesn't come up.

Reply:
This branch is for both being distinct, so:

```
SELECT DISTINCT a, b as c
FROM (SELECT DISTINCT a, b FROM t)
```

In this case, yes, the aliases are properly handled.

The branch on line 270 handles the `SELECT ... FROM (SELECT DISTINCT ...)` case. Right now it only works with `SELECT *` cases, but we might be able to make it work with outer queries that rename columns but otherwise select all of them. Right now I don't think that's worth it.